### PR TITLE
cmake: fix SDL2 linking

### PR DESCRIPTION
**Pull request by kmilos**
Fixes https://github.com/msys2/MINGW-packages/issues/12874

W/ some indentation/formatting changes...

Also for 4.0.1 if is to be built w/ more recent SDL2.

### CODE REVIEW

**Comment by TurboGit** [comment]
@kmilos : Can you double check that the merge on 4.0.x branch is ok as I had to resolve a conflict. TIA.

See 5bd343add948c7f326c4cd60297619cc05aaa0d5

File: src/libs/CMakeLists.txt [hunk 1]
```diff
@@ -138,19 +142,19 @@ endif(CUPS_FOUND)
...
-                              INSTALL_RPATH ${RPATH_ORIGIN}/../..)
-    endif(WIN32)
-    install(TARGETS ${module} DESTINATION ${CMAKE_INSTALL_LIBDIR}/darktable/plugins/lighttable COMPONENT DTApplication)
+  if(WIN32)
+    _detach_debuginfo (${module} ${CMAKE_INSTALL_LIBDIR}/darktable/plugins/lighttable)
+  else()
+    set_target_properties(${module}
+                          PROPERTIES
+                          INSTALL_RPATH ${RPATH_ORIGIN}/../..)
+  endif()
```
Comment (TurboGit) [inline src/libs/CMakeLists.txt:158]: BTW, now seeing that... It would have been better to keep the `endif(WIN32)` for clarity, no?

Comment (TurboGit) [reply src/libs/CMakeLists.txt:158]: Yes please, it is easier to find the matching `endif()` to me, so better keep this style. TIA.

**Comment by kmilos** [comment]
LGTM

**Pull request merged**